cannot find symbol       DoobooUtils.getInstance().rejectPromiseWithBillingError(promise, billingResult.getResponseCode());

### Version of react-native-iap

### Version of react-native

### Platforms you faced the error (IOS or Android or both?)

### Expected behavior

### Actual behavior

### Tested environment (Emulator? Real Device?)

### Steps to reproduce the behavior


After updating to 4.0 getting the error:

cannot find symbol
      DoobooUtils.getInstance().rejectPromiseWithBillingError(promise, billingResult.getResponseCode());

React Native version: 0.61.2
react-native-iap version: 4.0.0
Platform: Android, Android studio 3.4.2

Same error here, here is a bit more info

```
> Task :react-native-iap:compileDebugJavaWithJavac FAILED
/Users/home/Desktop/APLIKACE/test/node_modules/react-native-iap/android/src/main/java/com/dooboolab/RNIap/RNIapModule.java:506: error: cannot find symbol
      DoobooUtils.getInstance().rejectPromiseWithBillingError(promise, billingResult.getResponseCode());
                                                              ^
  symbol:   variable promise
  location: class RNIapModule
Note: /Users/home/Desktop/APLIKACE/test/node_modules/react-native-iap/android/src/main/java/com/dooboolab/RNIap/RNIapModule.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-iap:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

BUILD FAILED in 36s
252 actionable tasks: 225 executed, 27 up-to-date
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,8 @@
 ## Changelogs
-
+- **[4.0.1]**
+    - Resolve [#763](https://github.com/dooboolab/react-native-iap/issues/763)
+    - Resolve [#764](https://github.com/dooboolab/react-native-iap/issues/764)
+    - Resolve [#765](https://github.com/dooboolab/react-native-iap/issues/765)
 - **[4.0.0]**
   - rc1
     - Remove unused methods and fix typings

--- a/android/src/main/java/com/dooboolab/RNIap/RNIapModule.java
+++ b/android/src/main/java/com/dooboolab/RNIap/RNIapModule.java
@@ -110,7 +110,7 @@ public class RNIapModule extends ReactContextBaseJavaModule implements Purchases
             error.putString("code", errorData[0]);
             error.putString("message", errorData[1]);
             sendEvent(reactContext, "purchase-error", error);
-            DoobooUtils.getInstance().rejectPromiseWithBillingError(promise, billingResult.getResponseCode());
+            DoobooUtils.getInstance().rejectPromiseWithBillingError(PROMISE_BUY_ITEM, billingResult.getResponseCode());
           }
         }
       }
@@ -522,6 +522,9 @@ public class RNIapModule extends ReactContextBaseJavaModule implements Purchases
         item.putInt("purchaseStateAndroid", purchase.getPurchaseState());
 
         sendEvent(reactContext, "purchase-updated", item);
+      }
+      if (purchases.size() > 0) {
+        DoobooUtils.getInstance().resolvePromisesForKey(PROMISE_BUY_ITEM, purchases.get(0));
       }
     } else {
       WritableMap error = Arguments.createMap();

--- a/android/src/main/java/com/dooboolab/RNIap/RNIapModule.java
+++ b/android/src/main/java/com/dooboolab/RNIap/RNIapModule.java
@@ -503,7 +503,7 @@ public class RNIapModule extends ReactContextBaseJavaModule implements Purchases
       error.putString("code", errorData[0]);
       error.putString("message", errorData[1]);
       sendEvent(reactContext, "purchase-error", error);
-      DoobooUtils.getInstance().rejectPromiseWithBillingError(promise, billingResult.getResponseCode());
+      DoobooUtils.getInstance().rejectPromisesWithBillingError(PROMISE_BUY_ITEM, billingResult.getResponseCode());
       return;
     }
 

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "react-native-iap",
-  "version": "4.0.0",
+  "version": "4.0.1",
   "description": "React Native In App Purchase Module.",
   "main": "index.js",
   "types": "index.d.ts",

